cmd/apprepository-controller/main.go

```diff
@@ -30,29 +35,94 @@ import (
 	log "github.com/sirupsen/logrus"
 )
 
-var (
-	masterURL         string
-	kubeconfig        string
-	repoSyncImage     string
-	repoSyncCommand   string
-	namespace         string
-	dbURL             string
-	dbUser            string
-	dbName            string
-	dbSecretName      string
-	dbSecretKey       string
-	userAgentComment  string
-	crontab           string
-	reposPerNamespace bool
-)
+type arrayFlags []string
+
+// Config contains al the flags passed through the command line
+// besides, it contains the []LocalObjectReference for the PullSecrets
+// so they can be shared across the jobs
+type Config struct {
+	MasterURL                string
+	Kubeconfig               string
+	RepoSyncImage            string
+	RepoSyncImagePullSecrets arrayFlags
+	ImagePullSecretsRefs     []corev1.LocalObjectReference
+	RepoSyncCommand          string
+	Namespace                string
```

Comment:
I'd strongly prefer this to be `KubeappsNamespace` explicitly, as it gets confusing looking through code that refers to the namespaces of different resources etc.